eth/bor_api_backend.go

```diff
@@ -30,9 +30,56 @@ func (b *EthAPIBackend) GetRootHash(ctx context.Context, starBlockNr uint64, end
 	if err != nil {
 		return "", err
 	}
+
 	return root, nil
 }
 
+// GetRootHash returns root hash for given start and end block
+func (b *EthAPIBackend) GetVoteOnRootHash(ctx context.Context, starBlockNr uint64, endBlockNr uint64, rootHash string, milestoneId string) (bool, error) {
+	var api *bor.API
+	for _, _api := range b.eth.Engine().APIs(b.eth.BlockChain()) {
+		if _api.Namespace == "bor" {
+			api = _api.Service.(*bor.API)
+		}
+	}
+
+	if api == nil {
+		return false, errors.New("Only available in Bor engine")
+	}
+
+	localRootHash, err := api.GetRootHash(starBlockNr, endBlockNr)
+
+	if err != nil {
+		return false, err
+	}
+
+	if !b.eth.Miner().Mining() {
+		return localRootHash == rootHash, nil
+	}
+
+	downloader := b.eth.handler.downloader
+	isLocked := downloader.LockMutex(endBlockNr)
+
+	if !isLocked {
+		downloader.UnlockMutex(false, "", common.Hash{})
+
+		return false, errors.New("Previous sprint is still in locked state")
+	}
+
+	if localRootHash != rootHash[2:] {
+		downloader.UnlockMutex(false, "", common.Hash{})
+
+		return false, errors.New("RootHash mismatch+" + localRootHash + rootHash)
+	}
+
+	endBlock := b.eth.blockchain.GetBlockByNumber(endBlockNr)
+	endBlockHash := endBlock.Hash()
+
+	downloader.UnlockMutex(true, milestoneId, endBlockHash)
```

Comment:
we shouldn't have methods like that. it's https://medium.com/hackernoon/dancing-with-go-s-mutexes-92407ae927bf

could you find another solution with no explicit calls for Lock/Unlock in external types?